plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -147,11 +147,20 @@ export type SearchBarProps = Partial<SearchBarBaseProps>;
  * @public
  */
 export const SearchBar = ({ onChange, ...props }: SearchBarProps) => {
+  const configApi = useApi(configApiRef);
+  const analytics = useAnalytics();
   const { term, setTerm } = useSearch();
 
   const handleChange = (newValue: string) => {
-    setTerm(newValue);
-    if (onChange) onChange(newValue);
+    if (configApi.getOptionalConfig('app.analytics.ga')) {
```

Comment:
No need to check for this manually, the Analytics API will detect and forward events to the configured analytics tool automatically (whether it's GA or something else).

Reply:
yes thanks, updated after @Rugvip's comment!